conda_build/jinja_context.py

```diff
@@ -167,6 +167,43 @@ def load_npm():
         return json.load(pkg)
 
 
+def load_file_regex(config, load_file=None, regex_pattern=None, from_recipe_dir=False,
+                    recipe_dir=None, permit_undefined_jinja=True):
+    import re
+    match = False
+
+    cd_to_work = False
+
+    if load_file is None:
```

Comment:
This should maybe be an assert instead.  I'd also change load_file to not have a default argument, since it really is required.

```
assert load_file, 'File to be searched is not specified.'
```

On the other hand, raising the right exception is also important, and for this I'd probably think more about a ValueError than a RuntimeError.  This may or may not gel well will conda-build's existing exception handling.  My opinion is more to do things right here, and fix whatever else conda-build may not have done right elsewhere.